apps/storage/src/app.ts

```diff
@@ -0,0 +1,565 @@
+  const proxy = c.req.param('proxy');
+  const res = await fetch(
+    `${env.STORAGE_S3_ENDPOINT}/${env.STORAGE_S3_BUCKET_AVATARS}/${proxy}`
+  );
+  if (res.status === 404) {
+    return c.json({ error: 'Not Found' }, { status: 404 });
+  }
+  // Copy response as fetch headers are immutable
+  return new Response(res.body, res);
+});
+
+app
+  // Health check endpoint
+  .get('/', (c) => c.json({ status: "I'm Alive 🏝️" }))
+  // Attachment download endpoint
+  .get(
+    '/attachment/:orgShortCode/:attachmentId/:filename',
+    zValidator(
+      'param',
+      z.object({
+        filename: z.string().transform((f) => decodeURIComponent(f)),
+        attachmentId: typeIdValidator('convoAttachments'),
+        orgShortCode: z.string()
+      })
+    ),
+    async (c) => {
+      const { filename, attachmentId, orgShortCode } = c.req.valid('param');
+
+      const attachmentQueryResponse = await db.query.convoAttachments.findFirst(
+        {
+          where: eq(convoAttachments.publicId, attachmentId),
+          columns: {
+            fileName: true,
+            orgId: true,
+            public: true
+          }
+        }
+      );
+
+      if (
+        !attachmentQueryResponse ||
+        attachmentQueryResponse.fileName !== filename
+      ) {
+        return c.json(
+          { error: `Attachment ${filename} not found` },
+          { status: 404 }
+        );
+      }
+
+      const orgQueryResponse = await db.query.orgs.findFirst({
+        where: eq(orgs.shortcode, orgShortCode),
+        columns: {
+          id: true,
+          publicId: true
+        }
+      });
+
+      if (!orgQueryResponse) {
+        return c.json({ error: 'Invalid org' }, { status: 400 });
+      }
+
+      if (!attachmentQueryResponse.public) {
+        const accountId = c.get('account')?.id;
+        if (!accountId)
+          return c.json({ error: 'Unauthorized' }, { status: 401 });
+
+        const orgAccountMembershipResponse =
+          await db.query.orgMembers.findFirst({
+            where: and(
+              eq(orgMembers.orgId, orgQueryResponse.id),
+              eq(orgMembers.accountId, accountId)
+            ),
+            columns: {
+              id: true
+            }
+          });
+        if (!orgAccountMembershipResponse)
+          return c.json({ error: 'Unauthorized' }, { status: 401 });
+      }
+
+      const orgPublicId = orgQueryResponse.publicId;
+      const command = new GetObjectCommand({
+        Bucket: env.STORAGE_S3_BUCKET_ATTACHMENTS,
+        Key: `${orgPublicId}/${attachmentId}/${filename}`
+      });
+      const url = await getSignedUrl(s3Client, command, { expiresIn: 3600 });
+      const res = await fetch(url);
+      if (!res.ok) {
+        return c.json(
+          { error: 'Error while fetching attachment' },
+          { status: 500 }
+        );
+      }
+      // Copy response as fetch headers are immutable
+      return new Response(res.body, res);
+    }
+  )
+  // Avatar upload endpoint
+  .post(
```

Comment:
same here
lets not chain
makes it easier to parse and move things around later